File: src/main/scala/com/github/kazuhito_m/twitterbattler/primitive/view/SignupController.scala
package com.github.kazuhito_m.twitterbattler.primitive.view

import com.github.kazuhito_m.twitterbattler.primitive.infrastructure.security.AuthUtil
import org.slf4j.{Logger, LoggerFactory}
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.social.connect.web.ProviderSignInUtils
import org.springframework.social.connect.{ConnectionFactoryLocator, UsersConnectionRepository}
import org.springframework.stereotype.Controller
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.context.request.WebRequest

@Controller
class SignupController {

  private val log: Logger = LoggerFactory.getLogger(classOf[SignupController])

  private var signInUtils: ProviderSignInUtils = null

  @Autowired
  def this(
            connectionFactoryLocator: ConnectionFactoryLocator
            , connectionRepository: UsersConnectionRepository) {
    this()
    signInUtils = new ProviderSignInUtils(connectionFactoryLocator, connectionRepository)
  }

  @RequestMapping(value = Array("/signup"))
  def signup(request: WebRequest): String = {
    //    log.info("signup() が呼ばれた時。")
    //    log.info("signInUtils : " + signInUtils)
    val connection = signInUtils.getConnectionFromSession(request)
    if (connection != null) {
      AuthUtil.authenticate(connection)
      signInUtils.doPostSignUp(connection.getDisplayName, request)
    }
    "redirect:/"
  }

  @RequestMapping(value = Array("/signin"))
  def signin(request: WebRequest) = "redirect:/login.html"

}

File: server/src/main/java/com/github/kazuhito_m/twitterbattler/primitive/presentation/controller/signup/SignupController.java
package com.github.kazuhito_m.twitterbattler.primitive.presentation.controller.signup;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.context.request.WebRequest;

import java.beans.FeatureDescriptor;

@Controller
public class SignupController {
    final ConnectionFactoryLocator connectionFactoryLocator;
    final UsersConnectionRepository connectionRepository;

    private static final Logger LOGGER = LoggerFactory.getLogger(SignupController.class);

    ProviderSignInUtils signInUtils = new ProviderSignInUtils();

    @GetMapping("/signup")
    String signup(WebRequest request) {
        //    log.info("signup() が呼ばれた時。")
        //    log.info("signInUtils : " + signInUtils)
        FeatureDescriptor connection = signInUtils.getConnectionFromSession(request);
        if (connection != null) {
            AuthUtil.authenticate(connection);
            signInUtils.doPostSignUp(connection.getDisplayName(), request);
        }
        return "redirect:/";
    }

    @PostMapping("/signin")
    String signin(WebRequest request) {
        return "redirect:/login.html";
    }

}
